### PR TITLE
BUG: Convert color space before saving

### PR DESCRIPTION
**Pull request by MartinThoma**
The PDF contained an image in PA mode:

* P: 8-bit pixels, mapped to any other mode using a color palette
* PA: P with alpha

See #1801

### CODE REVIEW

**Comment by pubpub-zz** [comment]
is the PNG the best choice or should we use JPEG2000?

**Comment by MartinThoma** [comment]
@pubpub-zz I don't know where. But with my fix it looks good now :-)

**Comment by MartinThoma** [comment]
Maybe I should add some extracted images + make a visual diff :thinking: It's currently too easy to silently break something here

File: pypdf/filters.py [hunk 1]
```diff
@@ -683,6 +683,23 @@ def _xobj_to_image(x_object_obj: Dict[str, Any]) -> Tuple[Optional[str], bytes]:
...
+            elif color_space is not None and color_space[0] == "/ICCBased":
+                # see Table 66 - Additional Entries Specific to an ICC Profile
+                # Stream Dictionary
+                icc_profile = color_space[1].get_object()
+                color_components = cast(int, icc_profile["/N"])
+                alternate_colorspace = icc_profile["/Alternate"]
+                color_space = alternate_colorspace
+                mode_map = {
+                    "/DeviceGray": "L",
+                    "/DeviceRGB,": "RGB",
```
Comment (pubpub-zz) [inline pypdf/filters.py:695]: should not it be :
`"/DeviceRGB": "RGB",`

?

Comment (MartinThoma) [reply pypdf/filters.py]: Yes :sweat_smile:

**Pull request merged**